### PR TITLE
Major Refactor

### PR DESCRIPTION
**Pull request by nolram88**
Updated all files into import export model to reveal dependencies.
 This is stage one of the full refactor.
The next is to shred the larger files into smaller ones.

### CODE REVIEW

File: src/characters/baseActionStates.js [hunk 1]
```diff
@@ -1,8 +1,29 @@
+import {player, cS, drawVfx, screenShake, percentShake, finishGame, palettes, pPal,stage} from "../main/main";
+import {sounds} from "../main/sfx";
```
Comment (wbobeirne) [inline src/characters/baseActionStates.js]: Can you fix paths like these to not be relative? Webpack resolves at the root of src, so you can just do `main/sfx` etc.

Comment (nolram88) [inline src/characters/baseActionStates.js]: yea i gotta reconfigure webstorm. ill get right on it

File: bin/webpack/createConfig.js [hunk 2]
```diff
@@ -2,7 +2,9 @@ function createConfig(options) {
   const path = require("path");
   const webpack = require("webpack");
   const WebpackNotifierPlugin = require("webpack-notifier");
+  var HappyPack = require('happypack');
```
Comment (wbobeirne) [inline bin/webpack/createConfig.js]: Nice! Happypack definitely gives a solid speed boost. I've had some issues in the past with non-deterministic builds (File changes despite no actual changes,) but out build is simple enough that I don't think this'll be an issue.

File: src/characters/shared/moves/index.js [hunk 3]
```diff
@@ -1,81 +1,81 @@
...
+import OTTOTTO from "characters/shared/movesOTTOTTO";
+import OTTOTTOWAIT from "characters/shared/movesOTTOTTOWAIT";
+import MISSFOOT from "characters/shared/movesMISSFOOT";
+import FURASLEEPSTART from "characters/shared/movesFURASLEEPSTART";
+import FURASLEEPLOOP from "characters/shared/movesFURASLEEPLOOP";
+import FURASLEEPEND from "characters/shared/movesFURASLEEPEND";
+import STOPCEIL from "characters/shared/movesSTOPCEIL";
+import TECHU from "characters/shared/movesTECHU";
+import SLEEP from "characters/shared/movesSLEEP";
+import ENTRANCE from "characters/shared/movesENTRANCE";
```
Comment (wbobeirne) [inline src/characters/shared/moves/index.js]: Looks like these paths are messed up!

Comment (sheaf) [inline src/characters/shared/moves/index.js]: Good catch.

File: bin/webpack/createConfig.js (hunk 2 above)
Comment (nolram88) [inline bin/webpack/createConfig.js]: yea i was skeptical as well but it was super easy to set up and cut the build time in half for me. seems good so far

File: src/main/loadscreen.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 /* globals start */
 
 import $ from "jquery";
+import * as main from "main/main";
```
Comment (wbobeirne) [inline src/main/loadscreen.js]: You need to cut this, it's in a different bundle that this file loads. Causes main to execute twice.

File: src/main/loadscreen.js [hunk 5]
```diff
@@ -56,7 +57,7 @@ function loadScript() {
         loadScript();
       } else {
         loading = false;
-        start();
+        main.start();
```
Comment (wbobeirne) [inline src/main/loadscreen.js]: Change this to `window.start`, the main bundle attaches the start function to the window which this should execute. It's unfortunately the only way we have right now to communicate across bundles.

**Pull request merged**